packages/cli/src/event-source.ts

```diff
@@ -32,7 +35,16 @@ export const openEventSource = () => {
 			newEvent.type === 'new-input-props' ||
 			newEvent.type === 'new-env-variables'
 		) {
-			window.location.reload();
+			if (window.unsavedProps) {
+				notificationCenter.current?.addNotification({
+					id: 'random',
+					content: 'Do not save hey',
```

Comment:
- The content could be "There are unsaved changes to the props. Do you want to leave?"
- The notification center will do nothing here, you can only prevent a page from closing if you use `window.oneforeunload`

Example:
```
window.addEventListener('beforeunload', (event) => {
  event.returnValue = 'Are you sure you want to leave?';
});
```